src/components/advanced-table/advanced-table.tsx

```diff
@@ -131,14 +131,26 @@ export function AdvancedTable<
       })
       .then(payload => {
         setLoading(false);
+        const newRows = props.extractRows(payload) || [];
+
+        if (newRows.length === 0) {
+          // no rows in next page -- leave page unchanged, disable next page
+          // button, and show an error banner
+          setNextToken(undefined);
+          setDisplayError(trans.__('Last page reached.'));
```

Comment:
```suggestion
          setDisplayError(trans.__('This is the last page.'));
```

This may seem contradictory (why display a "next" button otherwise?) but it might be less jarring to see.

Reply:
Also, could this be somehow less severe than "error"? A less severe [alert](https://mui.com/material-ui/react-alert/), even "Info" (with state set in `setDisplayInfo`) would be more calming.

Reply:
1. I'd prefer to keep the string as is; it's more concise and conveys the same message.
2. I consider this an error, given that this should not happen for a backend correctly implementing `next_token`.